jbarlow83/OCRmyPDF
# WIP: Python API

Related issue #85.

I have started to refactor `__main__` and extracted `run_pipline` from the argument parsing.
This should not change the behavior (tests are working).

I would like some input before I continue with this feature.

**How should we validate the options of the api?**

`argparse` validates the names and enforces the types. If we add a python API the options should also be validated. I did not find a clean way to reuse the argparse configuration. Do we need to add an additional `check_options_keys` and `check_options_types` and duplicate the validation?

**Signature of the API?**

For now the signature is `ocrmypdf.run_pipline(Object)`.

I would suggest something like `ocrmypdf.run(input_file='some.pdf', output_file='out.pdf', ....)`.
If we stick to the names used in the argparse configuration we can reuse most of the code.
